cmd_manager.py

```diff
@@ -160,25 +165,69 @@ async def shop(self, ctx):
                 await shop_msg.add_reaction(arrow)
 
     @disc_cmds.command(name='timer')
-    async def timer(self, ctx, duration: int, unit='sec', name='', *args):
-        if len(args) > 0:
-            await ctx.send(content='Invalid number of arguments, please try again.')
-            return None
+    async def timer(self, ctx, *args):
+        message = ' '.join(args)
 
-        sec_duration = duration
-        if unit in ['min', 'minute', 'minutes', 'm']:
-            sec_duration = duration * 60
-        elif unit in ['hr', 'hour', 'hours', 'h']:
-            sec_duration = duration * 3600
-        elif unit in ['day', 'days', 'd']:
-            sec_duration = duration * 86400
-        elif unit not in ['sec', 'second', 'seconds', 's']:
-            await ctx.send(content='Invalid time unit specified, please try again.')
+        # Create a Calendar instance for time parsing
+        cal = parsedatetime.Calendar()
+
+        # Parse the message and obtain time delta
+        curr_time = datetime.now()
+        # res format: (datetime, result_code, start_idx, end_idx, matched_string)
+        res = cal.nlp(message)
+
+        # If the nlp returned None type, the parser failed to parse the time
+        if not res:
+            await ctx.send(content='Failed to parse time duration. Please try again.')
             return None
 
-        padded_name = '\"' + name + (' ' if name else '') + '\"'
+        res = res[0]
+        time_diff = res[0] - curr_time
+        # Offset the lost second due to processing
+        sec_duration = time_diff.total_seconds() + 1
```

Comment:
Not a fan of hard-coded offsets like this, but we can leave it in for now. Additionally, would it make sense to subtract instead of add because the user expects the timer to be from the time they submitted the timer not when we finish processing it?

Reply:
Necessary to display the correct time difference. Marking as resolved.